integration/test/performancetest/transmitter.go

```diff
@@ -153,12 +197,194 @@ func (transmitter *TransmitterAPI) TableExist() (bool, error) {
 /*
 SendItem()
 Desc: Parses the input data and adds it to the dynamo table
-Param: data []byte is the data collected by data collector
+Param: packet map[string]interface{} is the data collected by data collector
 */
 func (transmitter *TransmitterAPI) SendItem(packet map[string]interface{}) (string, error) {
-	//this is currently a passthrough function and will change to have functionality with Okan's PR
-	//@TODO
-	sentItem, err := transmitter.AddItem(packet)
+	// return nil
+	var sentItem string
+	var ae *types.ConditionalCheckFailedException // this exception represent the the atomic check has failed
+	// check if hash exists
+	currentItemList,err := transmitter.Query(packet[HASH].(string))
+	if err!=nil {
+		return "",err
+	}
+	if len(currentItemList)==0{ // if an item with the same hash doesn't exist add it
+		sentItem, err = transmitter.AddItem(packet) 
+		// this may be overwritten by other test threads, in that case it will return a specific error
+	
+		if !errors.As(err,&ae){ // check if our add call got overwritten by other threads
+			return sentItem,err
+		}
+		if err!=nil { //any other error dont try again
+			return "",err
+		}
+		// got overwritten now instead of adding go update the item
+		rand.Seed(time.Now().UnixNano())
+		time.Sleep(time.Duration(rand.Intn(UPDATE_DELAY_THRESHOLD))*time.Millisecond)
+		fmt.Println("Item already exist going to update",len(currentItemList))
+		
+	}
+	// item already exist so update the item instead
+	for {//concurrency retry
+		/*solving parallel updates using an optimistic lock and retry,
+		this may result in a livelock;
+		however, random sleeps makes this possiblity nearly impossible*/
+		currentItemList,err = transmitter.Query(packet[HASH].(string)) //get the item's most up to date version
+		newAttributes,err := transmitter.TestCasePackager(packet,currentItemList[0])
+		if(err!=nil){
+			return "",err
+		}
+		err = transmitter.UpdateItem(packet[HASH].(string),newAttributes) //try to update the item
+		//this may be overwritten by other test threads, in that case it will return a specific error
+		if errors.As(err,&ae){ //check if our call got overwritten
```

Comment:
Would be better to create another general function for handling optimism lock for DynamoDB